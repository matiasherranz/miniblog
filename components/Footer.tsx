import React, { FC, ReactElement } from 'react';
import Typography from '@material-ui/core/Typography';
import MuiLink from '@material-ui/core/Link';

const Footer: FC = (): ReactElement => {
  return (
    <Typography variant="body2" color="textSecondary" align="center">
      {'Made by '}
      <MuiLink color="inherit" href="https://linkedin.com/in/matiasherranz">
        Mat√≠as Herranz
      </MuiLink>
      {' using NextJS, Typescript and MaterialUI'}
      {'. | '}
      {new Date().getFullYear()}
    </Typography>
  );
};

export default Footer;
