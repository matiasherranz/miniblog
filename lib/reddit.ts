import axios from 'axios';
import { createAsyncThunk } from '@reduxjs/toolkit';

import { RedditStateType } from '../util/types';
import { BASE_REDDIT_URL, PAGE_SIZE } from '../util/constants';

type ParamsType = Record<string, string>;

export const fetchRedditPosts = createAsyncThunk<
  Promise<string>,
  void,
  { state: RedditStateType }
>(
  'notes/fetchPosts',
  async (_, { getState, rejectWithValue }) => {
    console.log('FETCHING POSTS');
    try {
      const { reddit } = getState();
      const params: ParamsType = {
        limit: PAGE_SIZE.toString(),
      };

      if (reddit.after) {
        params.after = reddit.after;
      }

      if (!reddit.hasMore) {
        throw new Error('No more posts available');
      }

      const response = await axios.get(`${BASE_REDDIT_URL}/top.json`, {
        params,
      });
      return response.data;
    } catch (error) {
      return rejectWithValue({ error: error.message });
    }
  },
  {
    condition: (_, { getState }) => {
      const { reddit } = getState();

      // Already fetching, cancel this new request
      return reddit.loading !== 'loading';
    },
  }
);
