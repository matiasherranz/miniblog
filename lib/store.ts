import { configureStore } from '@reduxjs/toolkit';
import { persistStore } from 'redux-persist';
import ReduxThunk from 'redux-thunk';

import redditReducer from './slices/redditSlice';

export const store = configureStore({
  reducer: {
    reddit: redditReducer,
  },
  devTools: true,
  middleware: [ReduxThunk],
});

export const persistor = persistStore(store);
