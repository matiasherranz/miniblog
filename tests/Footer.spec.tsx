import { mount } from 'enzyme';
import { Provider } from 'react-redux';

import { store } from '../lib/store';
import Footer from '../components/Footer';

describe('Footer', () => {
  const getWrapper = () =>
    mount(
      <Provider store={store}>
        <Footer />
      </Provider>
    );

  test('Footer component renders properly', () => {
    const wrapper = getWrapper();

    expect(wrapper).toMatchSnapshot();
  });
});
