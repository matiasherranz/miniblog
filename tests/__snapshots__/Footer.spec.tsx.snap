// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Footer component renders properly 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <Footer>
    <WithStyles(ForwardRef(Typography))
      align="center"
      className="makeStyles-footerMargin-1"
      color="textSecondary"
      variant="body2"
    >
      <ForwardRef(Typography)
        align="center"
        className="makeStyles-footerMargin-1"
        classes={
          Object {
            "alignCenter": "MuiTypography-alignCenter",
            "alignJustify": "MuiTypography-alignJustify",
            "alignLeft": "MuiTypography-alignLeft",
            "alignRight": "MuiTypography-alignRight",
            "body1": "MuiTypography-body1",
            "body2": "MuiTypography-body2",
            "button": "MuiTypography-button",
            "caption": "MuiTypography-caption",
            "colorError": "MuiTypography-colorError",
            "colorInherit": "MuiTypography-colorInherit",
            "colorPrimary": "MuiTypography-colorPrimary",
            "colorSecondary": "MuiTypography-colorSecondary",
            "colorTextPrimary": "MuiTypography-colorTextPrimary",
            "colorTextSecondary": "MuiTypography-colorTextSecondary",
            "displayBlock": "MuiTypography-displayBlock",
            "displayInline": "MuiTypography-displayInline",
            "gutterBottom": "MuiTypography-gutterBottom",
            "h1": "MuiTypography-h1",
            "h2": "MuiTypography-h2",
            "h3": "MuiTypography-h3",
            "h4": "MuiTypography-h4",
            "h5": "MuiTypography-h5",
            "h6": "MuiTypography-h6",
            "noWrap": "MuiTypography-noWrap",
            "overline": "MuiTypography-overline",
            "paragraph": "MuiTypography-paragraph",
            "root": "MuiTypography-root",
            "srOnly": "MuiTypography-srOnly",
            "subtitle1": "MuiTypography-subtitle1",
            "subtitle2": "MuiTypography-subtitle2",
          }
        }
        color="textSecondary"
        variant="body2"
      >
        <p
          className="MuiTypography-root makeStyles-footerMargin-1 MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-alignCenter"
        >
          Made by 
          <WithStyles(ForwardRef(Link))
            color="inherit"
            href="https://linkedin.com/in/matiasherranz"
          >
            <ForwardRef(Link)
              classes={
                Object {
                  "button": "MuiLink-button",
                  "focusVisible": "Mui-focusVisible",
                  "root": "MuiLink-root",
                  "underlineAlways": "MuiLink-underlineAlways",
                  "underlineHover": "MuiLink-underlineHover",
                  "underlineNone": "MuiLink-underlineNone",
                }
              }
              color="inherit"
              href="https://linkedin.com/in/matiasherranz"
            >
              <WithStyles(ForwardRef(Typography))
                className="MuiLink-root MuiLink-underlineHover"
                color="inherit"
                component="a"
                href="https://linkedin.com/in/matiasherranz"
                onBlur={[Function]}
                onFocus={[Function]}
                variant="inherit"
              >
                <ForwardRef(Typography)
                  className="MuiLink-root MuiLink-underlineHover"
                  classes={
                    Object {
                      "alignCenter": "MuiTypography-alignCenter",
                      "alignJustify": "MuiTypography-alignJustify",
                      "alignLeft": "MuiTypography-alignLeft",
                      "alignRight": "MuiTypography-alignRight",
                      "body1": "MuiTypography-body1",
                      "body2": "MuiTypography-body2",
                      "button": "MuiTypography-button",
                      "caption": "MuiTypography-caption",
                      "colorError": "MuiTypography-colorError",
                      "colorInherit": "MuiTypography-colorInherit",
                      "colorPrimary": "MuiTypography-colorPrimary",
                      "colorSecondary": "MuiTypography-colorSecondary",
                      "colorTextPrimary": "MuiTypography-colorTextPrimary",
                      "colorTextSecondary": "MuiTypography-colorTextSecondary",
                      "displayBlock": "MuiTypography-displayBlock",
                      "displayInline": "MuiTypography-displayInline",
                      "gutterBottom": "MuiTypography-gutterBottom",
                      "h1": "MuiTypography-h1",
                      "h2": "MuiTypography-h2",
                      "h3": "MuiTypography-h3",
                      "h4": "MuiTypography-h4",
                      "h5": "MuiTypography-h5",
                      "h6": "MuiTypography-h6",
                      "noWrap": "MuiTypography-noWrap",
                      "overline": "MuiTypography-overline",
                      "paragraph": "MuiTypography-paragraph",
                      "root": "MuiTypography-root",
                      "srOnly": "MuiTypography-srOnly",
                      "subtitle1": "MuiTypography-subtitle1",
                      "subtitle2": "MuiTypography-subtitle2",
                    }
                  }
                  color="inherit"
                  component="a"
                  href="https://linkedin.com/in/matiasherranz"
                  onBlur={[Function]}
                  onFocus={[Function]}
                  variant="inherit"
                >
                  <a
                    className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorInherit"
                    href="https://linkedin.com/in/matiasherranz"
                    onBlur={[Function]}
                    onFocus={[Function]}
                  >
                    Mat√≠as Herranz
                  </a>
                </ForwardRef(Typography)>
              </WithStyles(ForwardRef(Typography))>
            </ForwardRef(Link)>
          </WithStyles(ForwardRef(Link))>
           using NextJS, Typescript and MaterialUI
          . | 
          2020
        </p>
      </ForwardRef(Typography)>
    </WithStyles(ForwardRef(Typography))>
  </Footer>
</Provider>
`;
